src/lib/ledger_catchup/ledger_catchup.ml

```diff
@@ -184,49 +71,53 @@ module Make (Inputs : Inputs.S) :
            ~unprocessed_transition_cache verified_transition_with_hash
     in
     let open Deferred.Let_syntax in
-    match%map cached_verified_transition with
-    | Ok x -> Ok (Some x)
-    | Error `Duplicate ->
-        Logger.info logger ~module_:__MODULE__ ~location:__LOC__
+    match%bind cached_verified_transition with
+    | Ok x -> Deferred.return @@ Ok (Either.Second x)
+    | Error (`In_frontier hash) ->
+        Logger.trace logger ~module_:__MODULE__ ~location:__LOC__
           "transition queried during ledger catchup has already been seen" ;
-        Ok None
+        Deferred.return @@ Ok (Either.First hash)
+    | Error (`Under_processing consumed_state) -> (
+        Logger.trace logger ~module_:__MODULE__ ~location:__LOC__
+          "transition queried during ledger catchup is still under procession" ;
```

Comment:
"transition queried during ledger catchup is still in process in one of the componenets in the transition_frontier"